### PR TITLE
(chore) : clean up code following claims work

### PR DESCRIPTION
**Pull request by its-kios09**
## Requirements

- [ ] This PR has a title that briefly describes the work done, including the ticket number if there is a ticket.
- [ ] My work conforms to the [**OpenMRS 3.0 Styleguide**](https://om.rs/styleguide).
- [ ] I checked for feature overlap with [**existing widgets**](https://om.rs/directory).

## Summary

- Added Responsiveness of the claims page
- Added Metrics to provide Claims summary
- Added Filtering of the line items by PAID bill status
- Added a Custom Hook for fetching the providers and encounters
- Added Inline Loading and Inline notification for the end-user experience
- Added a functionality after submission to navigate back to the billing dashboard and a function to extract the name of the provider for better experience of the end-user
- Added validation of the form all fields are required
- Added itemOrBillableServiceConcept_id on the payload

## Screenshots
https://github.com/palladiumkenya/kenyaemr-esm-3.x/assets/67967749/cb4dc40d-bf40-4c1c-b1f1-0cb7276a21c8

*None.*

## Related Issue

*None.*

## Other

*None.*

### CODE REVIEW

File: packages/esm-billing-app/src/claims/dashboard/form/claims-form.component.tsx [hunk 1]
```diff
@@ -54,49 +57,51 @@ const ClaimsFormSchema = z.object({
...
+
+  const { data } = useProviders();
+  const [providers, setProviders] = useState([]);
+  const [loading, setLoading] = useState(false);
+
+  useEffect(() => {
+    if (data && data.data.results) {
+      setProviders(data.data.results.map((provider) => ({ id: provider.uuid, text: provider.display })));
+    }
+  }, [data]);
```
Comment (donaldkibet) [inline packages/esm-billing-app/src/claims/dashboard/form/claims-form.component.tsx:75]: no need for this `useEffect` we can instead just wait to the data to load for both visits and providers then display the form and transform the data to our desired format

**Review by donaldkibet — APPROVED** [review]
Thanks @its-kios09 cc @makombe

File: packages/esm-billing-app/src/claims/dashboard/form/claims-form.component.tsx (hunk 1 above)
Comment (its-kios09) [reply packages/esm-billing-app/src/claims/dashboard/form/claims-form.component.tsx]: The providers' array is directly mapped from the useProviders data and used within the FilterableMultiSelect component without needing an intermediate state.

File: packages/esm-billing-app/src/claims/claims-wrap/claims-main-component.tsx [hunk 2]
```diff
@@ -1,42 +1,34 @@
...
 
 interface ClaimsMainProps {
   bill: MappedBill;
 }
 
 const ClaimMainComponent: React.FC<ClaimsMainProps> = ({ bill }) => {
-  const { t } = useTranslation();
-
-  const [selectedLineItems, setSelectedLineItems] = useState([]);
+  const [selectedLineItems, setSelectedLineItems] = useState<LineItem[]>([]);
```
Comment (donaldkibet) [inline packages/esm-billing-app/src/claims/claims-wrap/claims-main-component.tsx:15]:
```suggestion
  const [selectedLineItems, setSelectedLineItems] = useState<Array<LineItem>>([]);
```

**Review by donaldkibet — APPROVED** [review]

**Pull request merged**